lib/ui/pages/qrgen_page.dart

```diff
@@ -0,0 +1,112 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:qr_flutter/qr_flutter.dart';
+import 'package:qrapp/models/downloadstatus.dart';
+import 'package:qrapp/services/qrservice.dart';
+
+class QRGenPage extends StatefulWidget {
+  const QRGenPage({Key? key}) : super(key: key);
+
+  @override
+  _QRGenPageState createState() => _QRGenPageState();
+}
+
+class _QRGenPageState extends State<QRGenPage> {
+  Timer? _timer;
+
+  @override
+  void initState() {
+    super.initState();
+    init();
+  }
+
+  init() async {
+    var provider = Provider.of<QRService>(context, listen: false);
+
+    await provider.getSeed(context);
+
+    _timer = Timer.periodic(const Duration(seconds: 1), (timer) {
+      setState(() {});
+    });
+
+    setState(() {});
```

Comment:
**question:** why are we calling setState here, which seems to be only called in the `initState`?